### DISCUSSION TITLE
Suddenly No Functions Can Run

### DISCUSSION THREAD

**Discussion by WilliamWelsh** [vercel/community#1203] [Help] [answered]
No config or deployments have happened in a few days, and all of a sudden none of my API routes are going through. Can anyone point me in the right direction? Everything works fine on my machine, and again nothing has changed since Dec 28, but suddenly this morning none of the API calls are going through. Even on my deployment site, which hasn't been updated in about a month, is throwing 500 internal server error.
![image](https://user-images.githubusercontent.com/16886888/210260585-20a280fd-9eec-4472-9f97-8c7f3547b0ff.png)

```
Child exited with error exit status 129START RequestId: 2a9f4fce-e3fe-4faf-8a64-d2648d3296f9 Version: $LATESTinfo  - Loaded env from /var/task/.env2023-01-02T16:41:35.013Z	2a9f4fce-e3fe-4faf-8a64-d2648d3296f9	ERROR	Uncaught Exception 	{"errorType":"Error","errorMessage":"Unexpected server response: 500","stack":["Error: Unexpected server response: 500","    at ClientRequest.<anonymous> (/var/task/node_modules/@ethersproject/providers/node_modules/ws/lib/websocket.js:604:7)","    at ClientRequest.emit (node:events:513:28)","    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:693:27)","    at HTTPParser.parserOnHeadersComplete (node:_http_common:117:17)","    at TLSSocket.socketOnData (node:_http_client:534:22)","    at TLSSocket.emit (node:events:513:28)","    at addChunk (node:internal/streams/readable:324:12)","    at readableAddChunk (node:internal/streams/readable:297:9)","    at Readable.push (node:internal/streams/readable:234:10)","    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)"]}Unknown application error occurredRuntime.Unknown
```

**Comment by amyegan** [comment]
Hi @WilliamWelsh. Do you have a public repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share? I have not seen this particular error before, and I'll need to do some digging to find the cause.

**Comment by aryan877** [comment]
having the same issue, any solutions

**Comment by stangatimu** [comment]
Hi also facing the same issue.
```The "emitter" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal```

**Comment by amyegan** [comment]
If anyone can share a minimal reproduction, I would be happy to take a closer look

**Comment by WilliamWelsh** [comment, marked as answer]
@aryan877 @stangatimu My issue was trying to use websockets inside of API routes. They weren't edge functions, and one day a new build just started throwing those errors. Back then I refactored to use regular http calls in my API routes instead of websockets.

**Reply by aryan877** [reply]
@WilliamWelsh @amyegan Got you, i was trying to use it on the browser, but my provider url was misconfigured for ethers with wagmi, I guess we had different reasons for the error

**Reply by WilliamWelsh** [reply]
Glad you figured it out, I was using ethers websocket RPCs and had to change to regular Json RPCs
